RosNet/RosReader/RosBagReader.cs

```diff
@@ -61,8 +83,257 @@ public static RosBag Read(string path)
             }
             else
             {
-                throw new FileNotFoundException("File does not exist");
+                throw new FileNotFoundException($"File with path {path} does not exist");
+            }
+        }
+
+        /// <summary>
+        /// Reads a connection and sets the data
+        /// </summary>
+        private static void SetConnectionData(BinaryReader reader, ref Connection connection)
+        {
+            int dataLen = reader.ReadInt32();
+            long endPos = reader.BaseStream.Position + dataLen;
+
+            byte[] originalTopic = null;
+            byte[] type = null;
+            byte[] md5sum = null;
+            List<FieldValue> messageDefinition = null;
+            byte[] callerID = null;
+            byte[] latching = null;
+            while (reader.BaseStream.Position != endPos)
+            {
+                int fieldLen = reader.ReadInt32();
+                string fieldName = ReadName(reader);
+                byte[] fieldValue = reader.ReadBytes(fieldLen - fieldName.Length - 1);
+
+                switch (fieldName)
+                {
+                    case "topic":
+                        originalTopic = fieldValue;
+                        break;
+                    case "type":
+                        type = fieldValue;
+                        break;
+                    case "md5sum":
+                        md5sum = fieldValue;
+                        break;
+                    case "message_definition":
+                        messageDefinition = MessageDefinitionParser.ParseMessageDefinition(fieldValue);
+                        break;
+                    case "callerid":
+                        callerID = fieldValue;
+                        break;
+                    case "latching":
+                        latching = fieldValue;
+                        break;
+                }
+
+            }
+            connection.SetData(originalTopic, type, md5sum, messageDefinition, callerID, latching);
+        }
+
+        /// <summary>
+        /// Reads message data
+        /// </summary>
+        /// <returns>message data in bytes</returns>
+        private static byte[] ReadMessageData(BinaryReader reader)
```

Comment:
Alternatively (as suggested by the guidelines), one should amke `RosBagReader` subclass `BinaryReader`, and implement the methods in non-static ways. Since `RosBagReader` is entierly internal either way is basically fine, but it appears the latter is preferred.

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods#general-guidelines

Reply:
Or just store the `BinaryReader` in the `RosBagReader` and make in non-static, which is also a better solution than this